ddp-workspace/projects/ddp-sdk/src/lib/components/languageSelector.component.ts

```diff
@@ -73,43 +85,134 @@ export class LanguageSelectorComponent implements OnInit, OnDestroy {
   }
 
   public changeLanguage(lang: StudyLanguage): void {
+    if (!isNullOrUndefined(this.currentLanguage) && this.currentLanguage.languageCode === lang.languageCode) {
+      return;
+    }
+
     if (this.language.canUseLanguage(lang.languageCode)) {
       this.currentLanguage = lang;
-      this.language.changeLanguage(lang.languageCode);
+      if (this.language.getCurrentLanguage() !== lang.languageCode) {
+        this.language.changeLanguage(lang.languageCode);
+        if (this.session.isAuthenticatedSession()) {
+          this.updateProfileLanguage();
+        }
+      }
     }
     else {
       console.error('Error: The specified language: ' + JSON.stringify(lang) + ' is not configured for the study.');
     }
   }
 
+  //Update the language in the profile to the current language
+  private updateProfileLanguage(): void {
+    // Create observable that gets the user's profile and creates a copy with updated preferred language
+    let getProfileObservable: Observable<UserProfile> = this.profileServiceAgent.profile
+      .pipe(map(x => {
+        return {...x.profile, preferredLanguage: this.currentLanguage.languageCode};
+      })).pipe(take(1));
+
+    // When we get the new profile value, save it
+    let updateProfileObservable: Observable<any> = getProfileObservable
+      .pipe(concatMap(profile => this.profileServiceAgent.saveProfile(false, profile)));
+
+    //Subscribe to the combined observable
+    let sub: Subscription = updateProfileObservable.subscribe(() => this.language.notifyOfProfileLanguageUpdate(true));
+    this.anchor.addNew(sub);
+  }
+
   //Find the current language and return true if successful or false otherwise
-  public findCurrentLanguage(): boolean {
-    //Check if we already have a current language
-    if (!isNullOrUndefined(this.currentLanguage)) {
-      this.changeLanguage(this.currentLanguage);
-      return true;
-    }
+  public findCurrentLanguage(): Observable<Boolean> {
```

Comment:
We should use here type instead of class.
```suggestion
  public findCurrentLanguage(): Observable<boolean> {
```

Reply:
Good point.  I forgot that TypeScript doesn't require you to use a class for generics like Java does...